libs/application/core/src/lib/formUtils.ts

```diff
@@ -101,27 +102,39 @@ export const getFormNodeLeaves = (node: FormNode): FormLeaf[] => {
   return leaves
 }
 
-export function getSectionsInForm(form: Form): Section[] {
+export function getSectionsInForm(
+  form: Form,
+  answers: FormValue,
+  externalData: ExternalData,
+): Section[] {
   const sections: Section[] = []
   form.children.forEach((child) => {
-    if (child.type === FormItemTypes.SECTION) {
+    const shouldShowSection = shouldShowFormItem(child, answers, externalData)
+    if (child.type === FormItemTypes.SECTION && shouldShowSection) {
       sections.push(child as Section)
     }
   })
   return sections
 }
-export function getSubSectionsInSection(section: Section): SubSection[] {
+export function getSubSectionsInSection(
+  section: Section,
+  answers: FormValue,
+  externalData: ExternalData,
+): SubSection[] {
   const subSections: SubSection[] = []
-  section.children.forEach((child) => {
-    if (child.type === FormItemTypes.SUB_SECTION) {
+  section?.children.forEach((child) => {
```

Comment:
if `section` can be undefined, then `forEach` will fail because of not being an array. Add extra check `(section?.children ?? [])`, isn't it?